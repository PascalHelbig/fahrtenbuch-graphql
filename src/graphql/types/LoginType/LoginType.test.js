const { makeExecutableSchema, addMockFunctionsToSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const LoginType = require('./LoginType');

it('has to be a valid schema', () => {
  const SchemaDefinition = 'schema { query: Login }';
  const schema = makeExecutableSchema({ typeDefs: [SchemaDefinition, LoginType] });
  addMockFunctionsToSchema({
    schema,
    mocks: {
      Login: () => ({
        token: () => 'the token',
        user: () => ({ email: 'test@example.com' }),
      }),
    },
  });

  const query = `
    query {
      token
      user { email }
    }
  `;

  return graphql(schema, query).then(result =>
    expect(result).toMatchSnapshot(),
  );
});
