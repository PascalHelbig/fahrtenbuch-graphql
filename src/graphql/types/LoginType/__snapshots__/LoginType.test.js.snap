exports[`test has to be a valid schema 1`] = `
Object {
  "data": Object {
    "token": "the token",
    "user": Object {
      "email": "test@example.com",
    },
  },
}
`;
